Blocks: Add fallback when no featured image is available

coreymckrill opened the pull request:
~Modifies both JS and PHP to render a fallback container with icon when a post does not have a featured image.~

Ensures that an empty string is output instead of a broken image tag when a post does not have a featured image available.

This also makes some improvements to the featured image output when there _is_ a featured image both on the JS and the PHP side.

These blocks also need to be updated to work with this:

- [x] Sessions
- [x] Sponsors

Refs #95

coreymckrill committed:
Blocks: Add fallback when no featured image is available

Modifies both JS and PHP to render a fallback container with icon
when a post does not have a featured image.

iandunn commented:
I started reviewing this on Friday, but now it sounds like [Mel doesn't think we should have a fallback](https://github.com/WordPress/wordcamp.org/issues/95#issuecomment-485530517), so I'll hold off doing anything else until.

coreymckrill committed:
Sessions: Revert changes that added a featured image fallback

coreymckrill committed:
FeaturedImage: Remove code that implements fallback

coreymckrill committed:
FeaturedImage: Remove unnecessary width units

In the `width` attribute of the `img` tag, the units are implicitly
pixels, so it is not necessary to include them.

coreymckrill committed:
FeaturedImage: Remove unused imports

coreymckrill committed:
Remove unnecessary stylesheet dependency

coreymckrill committed:
Remove another unused import

coreymckrill committed:
FeaturedImage: Refactor PHP render function to make classes optional

This moves the `$class_names` function parameter of
`render_featured_image` after the required parameters and adds a
default value to it, ensuring that it isn't necessary to include
a value for it when invoking the function.

The two blocks that currently use the function are also updated to
reflect the change in parameter order.

This also makes small improvements to the way that the container
classes are handled in the function.

coreymckrill commented:
Sounds good. I've changed the description here to reflect that this only ensures that _nothing_ is output when a featured image isn't available, instead of outputting a new fallback element.

coreymckrill committed:
FeaturedImage: Refactor to simplify usage of fetched data

This changes the inputs for the FeaturedImage component to simplify
them. Since several required data items come directly from the REST
API response, this allows us to feed all of that response data into the
`imageData` property, and then extract it from there, right in
the component, instead of extracting separate pieces elsewhere and
feeding them into multiple properties.

Notably, the image alt text is now extracted from the response data
instead of being defined externally. The Sessions and Sponsors blocks
are also updated to reflect this, because they were previously
incorrectly using the post title as the image alt text.

This also calculates the correct height for the image, based on the
set width and the aspect ratio, and adds the height attribute to the
`img` element. This way, the image dimensions are displayed correctly
without CSS rules that may collide with custom CSS for the site.

These changes bring the rendered HTML closer to what is produced
by the PHP function for the front end.

coreymckrill committed:
FeaturedImage: Minimize differences in HTML markup between JS and PHP

coreymckrill commented:
Committed here: https://meta.trac.wordpress.org/changeset/8668
